backend/app/mcp_server/tools/silent_exit.py

```
@@ -0,0 +1,93 @@
+            subtask.result["silent_exit_reason"] = reason
+            subtask.status = SubtaskStatus.COMPLETED
+            subtask.progress = 100
+            subtask.updated_at = datetime.now()
+            flag_modified(subtask, "result")
+            db.commit()
+            logger.info(f"[MCP] Marked subtask {subtask_id} as silent exit")
+        else:
+            logger.warning(f"[MCP] Subtask {subtask_id} not found for silent exit")
+    except Exception as e:
+        db.rollback()
+        logger.error(f"[MCP] Error updating subtask {subtask_id}: {e}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)